components/pkg-export-helm/src/values.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate habitat_pkg_export_docker as export_docker;
+
+use std::io::Write;
+
+use export_docker::Result;
+
+pub struct Values {
```

Comment:
Naming and typing questions... I was initially thinking that this was another case of using JSON types directly.

What's the benefit of this type versus a plain `HashMap`? Is insertion order important?

Reply:
I actually went with HashMap approach at first but I wanted to have custom methods on it that makes it simpler to add entries and then also a method to generate the file. I could just have this as a container of HashMap and get rid of ValuesEntry private type below but I'm a bit unsure if that approach is better than the current very explicit one. It'd only save us 4 LOCs above.